firebase-firestore/src/jsMain/kotlin/dev/gitlive/firebase/firestore/firestore.kt

```diff
@@ -257,7 +255,15 @@ actual class DocumentReference(val js: firebase.firestore.DocumentReference) {
     actual suspend fun update(vararg fieldsAndValues: Pair<String, Any?>) = rethrow {
         fieldsAndValues.takeUnless { fieldsAndValues.isEmpty() }
             ?.map { (field, value) -> field to encode(value, true) }
-            ?.let { encoded -> js.update(encoded.toMap()) }
```

Comment:
incorrect method was used for fields update. fixed by using the same approach `update(vararg fieldsAndValues: Pair<FieldPath, Any?>)` does